File: android/src/main/kotlin/co/quis/flutter_contacts/properties/Group.kt
package co.quis.flutter_contacts.properties

data class Group(
    var id: String,
    var name: String,
    var accountId: String
) {

    companion object {
        fun fromMap(m: Map<String, Any>): Group = Group(
            m["id"] as String,
            m["name"] as String,
            m["accountId"] as String,
        )
    }

    fun toMap(): Map<String, Any> = mapOf(
        "id" to id,
        "name" to name,
        "accountId" to accountId,

    )
}

File: ios/Classes/properties/Group.swift
import Contacts

@available(iOS 9.0, *)
struct Group {
    var id: String
    var name: String
    var accountId: String

    init(fromMap m: [String: Any]) {
        id = m["id"] as! String
        name = m["name"] as! String
        accountId = m["accountId"] as! String
    }

    init(fromGroup g: CNGroup, accountId: String) {
        id = g.identifier
        name = g.name
        self.accountId = accountId
    }

    func toMap() -> [String: Any] { [
        "id": id,
        "name": name,
        "accountId": accountId,
    ]
    }
}
